apps/client/src/Components/Buttons/ExternalLinkButton/index.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { FC, ButtonHTMLAttributes } from 'react'
+
+import { StyledButton } from './style'
+
+export interface ButtonProps {
+  href: string
+  buttonColor?: 'primary' | 'secondary' | 'secondary-variant-1' | 'white' | 'gray'
+}
+
+type ButtonType = ButtonHTMLAttributes<HTMLButtonElement>
+
+export const ExternalLinkButton: FC<ButtonProps & ButtonType> = ({
+  buttonColor = 'white',
+  href,
+  children,
+  ...props
+  }) => {
+  return (
+    <>
+      <a href={href} >
```

Comment:
essa estrutura funciona caso o usuário do componente queira extender a estilização do componente? isto é, o snippet de código abaixo funcionaria?

```typescript
import styled from "styled-components"
import ExternalLinkButton from "Components/Buttons/ExternalLinkButton"

const MyButton = styled(ExternalLinkButton)`
  background-color: var(--super-cool-new-color, #B00B69);
`
```

é importante que o usuário dos componentes possa expandir eles. Se isso não for possível pela relação do botão com o <a> e não puder ser mudado aí a gente faz esse sacrifício, mas se for possível manter essa extensibilidade eu acho bom

Reply:
Pelo que eu teste aqui, até aplica a cor nova como o exemplo, mas ele não está estendendo as estilizações que vem do ExternalLinkButton